src/overpass_api/data/collect_items.h

```diff
@@ -330,7 +331,7 @@ void collect_items_range(const Statement* stmt, Resource_Manager& rman,
       count = 0;
       rman.health_check(*stmt, 0, eval_map(result));
     }
-    if (predicate.match(Object::get_id(it.object_data())))
+    if (predicate.match(it.apply_func(&Object::get_id)))
```

Comment:
Some predicates like Get_Parent_Ways_Predicate need the full object to work properly. This currently not needed for *collect_items_range*, but we need a strong compile time check for it... 

Otherwise, '*:Id_Type` gets automatically converted to an `*_Skeleton` instance, and `match` operates on an incomplete object instance.